lib/utils/job-utils/http-tool.js

```diff
@@ -0,0 +1,202 @@
+//Copyright 2016, EMC, Inc.
+
+/*jshint node: true*/
+/*jshint multistr: true*/ 
+
+'use strict';
+
+var di = require('di');
+
+module.exports = HttpFactory;
+
+di.annotate(HttpFactory, new di.Provide('JobUtils.HttpTool'));
+di.annotate(HttpFactory, new di.Inject(
+    'Assert',
+    'Promise',
+    '_'
+));
+
+function HttpFactory(assert, Promise, _) {
+    
+    /**
+    * Tool class that does HTTP methods
+    * @constructor
+    */
+    function HttpTool(){
+        this.settings = {};
+        this.urlObject = {};
+        this.dataToWrite = '';
+        // ********** Helper Functions/Members **********
+        
+        this.validMethods = ['GET', 'PUT', 'POST', 'DELETE', 'PATCH'];
+        
+        /**
+        * Make sure that settings has at least property of url and method.
+        * @return {boolean} whether settings is valid.
+        */ 
+        this.isSettingValid = function(settings){
+            if (_.isEmpty(settings)) {
+                return false;
+            }
+            if ( ! ('url' in settings && 'method' in settings)) {
+                return false;
+            }
+            if (_.isEmpty(settings.url) || _.isEmpty(settings.method)) {
+                return false;
+            }
+            return true;
+        };
+
+        /**
+        * Parse and convert setting into a urlObject that suitable for 
+        * http/https module in NodeJs.
+        * @return {object} - the objec that suitable for Node http/https module.
+        */
+        this.setupUrlOptions = function(settings) {
+                
+            var urlTool = require('url');
+            var urlObject = {};
+
+            // Parse the string into url options
+            if (typeof (settings.url) === 'string') {
+                urlObject = urlTool.parse(settings.url);
+            }
+            else {
+                urlObject = settings.url;
+            }
+            // set the REST options
+            urlObject.method = settings.method;
+        
+            // set up the REST headers
+            if (! _.isEmpty(settings.headers)){
+                urlObject.headers = settings.headers;
+            }
+            else {
+                urlObject.headers = {};
+            }
+            
+            urlObject.headers['Content-Length'] = 0;
+            if (settings.hasOwnProperty('data')){
+                this.dataToWrite = JSON.stringify(settings.data);
+                urlObject.headers['Content-Length'] = Buffer.byteLength(this.dataToWrite);
+            }
+            
+            if (! _.isEmpty(settings.credential)){
+                urlObject.auth = settings.credential.username +':'+
+                    settings.credential.password;
```

Comment:
How about if either `username` or `password` is missing, I didn't see such validation logic.